src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java

```diff
@@ -794,6 +795,49 @@ public void testJavaParseExceptionSeverityDefaultError() throws Exception {
         verify(checker, files, expectedViolation);
     }
 
+    /**
+     * Verifies that TreeWalker sorting works correctly using AbstractCheck::getId.
+     * Multiple checks are configured with unique IDs to ensure all are executed and
+     * violations are reported in a sorted order.
+     */
+    @Test
+    public void testCheckSortingByIdWithVerify() throws Exception {
+        final DefaultConfiguration illegalCatchCheck = createModuleConfig(IllegalCatchCheck.class);
+        illegalCatchCheck.addAttribute("id", "id1");
```

Comment:
you should not have this configs, as all ofthem should be loaded from Input file